ocfweb/api/session_tracking.py

```diff
@@ -0,0 +1,112 @@
+import json
+from functools import partial
+from ipaddress import ip_address
+
+from django.conf import settings
+from django.http import HttpResponse
+from django.http import HttpResponseBadRequest
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+from ipware.ip import get_real_ip
+from ocflib.infra.hosts import hosts_by_filter
+from ocflib.infra.net import is_ocf_ip
+from ocflib.lab.stats import get_connection
+
+from ocfweb.caching import cache
+
+get_connection = partial(
+    get_connection,
+    user=settings.OCFSTATS_USER,
+    password=settings.OCFSTATS_PASSWORD,
+    db=settings.OCFSTATS_DB,
+)
+
+
+@require_POST
+@csrf_exempt
+def log_session(request):
+    """Primary API endpoint for session tracking.
+
+    Desktops have a cronjob that calls this endpoint, replacing
+    the functionality that used to be in ocf/labstats.
+    """
+
+    remote_ip = get_real_ip(request)
+
+    if not is_ocf_ip(ip_address(remote_ip)):
+        return HttpResponse('Not Authorized', status=401)
+
+    try:
+        body = json.loads(request.body.decode('utf-8'))
+
+        host = _get_desktops().get(remote_ip)
+        user = body.get('user')
+        state = body.get('state')
+
+        if not user:
+            raise ValueError('Invalid user "{}"'.format(user))
+
+        if host is None:
+            raise ValueError('Invalid host "{}"'.format(host))
```

Comment:
Maybe this should log the host IP instead?

Reply:
Agreed, this should log the IP, logging `None` is not useful.

Reply:
will fix